st2common/st2common/models/api/keyvalue.py

```diff
@@ -44,6 +53,16 @@ class KeyValuePairAPI(BaseAPI):
                 'type': 'string',
                 'required': True
             },
+            'secret': {
```

Comment:
Does `encrypted` only mean that whether the current display value or value returned from the API is encrypted? It as such provides no guarantee that it is stored in an encrypted manner in the DB since that is a function of `secret`.